compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -650,54 +657,51 @@ abstract class GenJSCode extends plugins.PluginComponent
         constructorTrees: List[DefDef]): js.Tree = {
       implicit val pos = classSym.pos
 
-      // Implementation restriction
-      val syms = constructorTrees.map(_.symbol)
-      val hasBadParam = enteringPhase(currentRun.uncurryPhase) {
-        syms.exists(_.paramss.flatten.exists(p => p.hasDefault))
-      }
-      if (hasBadParam) {
+      if (hasDefaultCtorArgsAndRawJSModule(classSym)) {
         reporter.error(pos,
-            "Implementation restriction: the constructor of a " +
-            "Scala.js-defined JS classes cannot have default parameters.")
-      }
-
-      withNewLocalNameScope {
-        val ctors: List[js.MethodDef] = constructorTrees.flatMap { tree =>
-          genMethodWithCurrentLocalNameScope(tree)
-        }
+            "Implementation restriction: constructors of " +
+            "Scala.js-defined JS classes cannot have default parameters " +
+            "if their module is JS native.")
+        js.EmptyTree
+      } else {
```

Comment:
Yes there was an issue, the compilation was crashing. If I remember well it was an assertion fail due to a malformed tree.